featuretools/primitives/utils.py

```diff
@@ -93,6 +97,72 @@ def list_primitives():
     return pd.concat([agg_df, transform_df], ignore_index=True)[columns]
 
 
+def summarize_primitives() -> pd.DataFrame:
+    """Returns a metrics summary DataFrame of all built-in primitives in featuretools."""
+    (
+        trans_names,
+        trans_primitives,
+        trans_valid_inputs,
+        trans_return_type,
+    ) = _get_names_primitives(get_transform_primitives)
+
+    (
+        agg_names,
+        agg_primitives,
+        agg_valid_inputs,
+        agg_return_type,
+    ) = _get_names_primitives(get_aggregation_primitives)
+
+    tot_trans = len(trans_names)
+    tot_agg = len(agg_names)
+    tot_prims = tot_trans + tot_agg
+    all_primitives = trans_primitives + agg_primitives
+    primitives_summary = _get_summary_primitives(all_primitives)
+    summary_dict = {
+        "total_primitives": tot_prims,
+        "aggregation_primitives": tot_agg,
+        "transform_primitives": tot_trans,
+        "unique_input_types": primitives_summary["unique_input_types"],
+        "unique_out_types": primitives_summary["unique_output_types"],
+        "use_multi_inputs": primitives_summary["ct_multi_in"],
+        "use_multi_outputs": primitives_summary["ct_multi_out"],
+        "use_external_data": primitives_summary["ct_extra_data"],
+        "are_controllable": primitives_summary["ct_controllable"],
+        "use_time_index": primitives_summary["ct_semantic_time_idx"],
+        "use_datetime_inputs": primitives_summary["datetime"],
+        "use_categorical_inputs": primitives_summary["categorical"],
+        "use_address_inputs": primitives_summary["address"],
+        "use_age_inputs": primitives_summary["age"],
+        "use_age_nullable_inputs": primitives_summary["age_nullable"],
+        "use_age_fractional_inputs": primitives_summary["age_fractional"],
+        "use_boolean_inputs": primitives_summary["boolean"],
+        "use_boolean_nullable_inputs": primitives_summary["boolean_nullable"],
+        "use_country_code_inputs": primitives_summary["country_code"],
+        "use_currency_code_inputs": primitives_summary["currency_code"],
+        "use_double_inputs": primitives_summary["double"],
+        "use_integer_inputs": primitives_summary["integer"],
+        "uses_integer_nullable_inputs": primitives_summary["integer_nullable"],
+        "use_email_address_inputs": primitives_summary["email_address"],
```

Comment:
what if we generated many of these entries using something like
`{f"use_{ltype}_inputs": count for ltype, count in primitives_summary["intputs"].items()}`